src/msg/async/Stack.h

```diff
@@ -259,12 +274,27 @@ class Worker {
 
     perf_logger = plb.create_perf_counters();
     cct->get_perfcounters_collection()->add(perf_logger);
+
+    // Add labeled perfcounters
+    std::string labels = ceph::perf_counters::key_create(
+      name_prefix, {{"id", std::to_string(id)}});
+    PerfCountersBuilder plb_labeled(cct, labels, l_msgr_labeled_first, l_msgr_labeled_last);
+
+    plb_labeled.add_u64_counter(l_msgr_connection_ready_timeouts, "msgr_connection_ready_timeouts", "Number of not yet ready connections declared as dead", NULL, PerfCountersBuilder::PRIO_USEFUL);
```

Comment:
formatting (I know that you just followed existing code for the unlabeled case but the labeled case is net new so let's not pile on)
```
    std::string labels = ceph::perf_counters::key_create(
        name_prefix, {{"id", std::to_string(id)}});
    PerfCountersBuilder plb_labeled(cct, labels, l_msgr_labeled_first,
                                    l_msgr_labeled_last);

    plb_labeled.add_u64_counter(
        l_msgr_connection_ready_timeouts, "msgr_connection_ready_timeouts",
        "Number of not yet ready connections declared as dead", NULL,
        PerfCountersBuilder::PRIO_USEFUL);
    plb_labeled.add_u64_counter(
        l_msgr_connection_idle_timeouts, "msgr_connection_idle_timeouts",
        "Number of connections closed due to idleness", NULL,
        PerfCountersBuilder::PRIO_USEFUL);
```